### ISSUE TITLE
Documentation of data format

### ISSUE THREAD

**Issue by brockho** [numbbo/coco#1381]
Following the discussion on #506, it is important to update our documentation of the COCO output format. Some specification details have changed since the [original document](http://coco.lri.fr/downloads/download15.03/bbobdocexperiment.pdf) (see appendix E).

**Comment by nikohansen** [comment]
Maybe one "additional" specification: the first line in the `.info` file for each run, which reads for example 
```
funcId = 10, DIM = 20, Precision = 1.000e-08, algId = 'random_search'
```
should become valid Python code when surrounded like ``"dict(" + line + ")"``. For safety concerns such code however should never be executed.

**Comment by brockho** [comment]
FTR, the LaTeX file to produce the documentation is in our svn repository under `coco/BBOB/doc/experimentation/`

**Comment by nikohansen** [comment]
See also #1160, where annotation of columns is suggested (which is probably superseeded by the `TestBed` class).

**Comment by nikohansen** [comment]
The observer/logger should write its name in the info file (as now the suite name is written).

**Comment by nikohansen** [comment]
Related issue on the format we need in the constrained case: #1367

**Comment by brockho** [comment]
Today, we decided upon the best place for the documentation. After examining the pros and cons of several options (wiki, howtos folder, coco-doc repository, ...), @nikohansen had the idea that the logger, in the long run, should actually write the documentation into every experiments output folder. For this, a  `data_format_specification.c` file within `code-experiments/src/` will be needed.

As a first step to start documenting, we can start by writing a text file within the `code-experiments/documentation/` folder.

**Comment by nikohansen** [comment]
This issue is blocking issues #1444 and #1558, because the latter need to abide by the correct specification.

**Comment by nikohansen** [comment]
The documentation should also mention why `cocopp` depends on a first line written with evaluation counter `1`.

**Comment by nikohansen** [comment]
My suggestion is that the `bbob` logger will in future invariably write a constraints evaluation column. We may call the data format `bbob` or `bbob2` or `bbob-constrained`. Remark that the current bbob logger doesn't write a data format specification. We could call it `bbob0` or `bbob1` or `bbob`.

**Comment by asmaatamna** [comment]
Something is not clear to me: if the logger now systematically writes a "g evaluations" column, why do we need two data formats? (```bbob0``` and ```bbob```)

**Comment by brockho** [comment]
Data, written in the old format, is available to the public (and will be for long probably). We should therefore also document the old format, even if the current code is not writing it anymore (but the old versions will).

**Comment by dtusar** [comment]
The documentation is written in https://github.com/numbbo/coco/blob/development/code-experiments/documentation/data_format.txt.

> The documentation should also mention why cocopp depends on a first line written with evaluation counter `1`.

We have this line 

`assert all(self.evals[0][1:] == 1)`

in `pproc.py` which demands that. I removed this line and tried to run with data that started with number `2`. The post-processing produced all plots and tables and I didn't see any notable change in the output (compared to the same data but with the first evaluation starting with `1`). This `assert` is in the code from 2013. Maybe at that time we needed it but I don't see any reason to still keep it.

**Comment by nikohansen** [comment]
> I don't see any reason to still keep it.

I don't see how you could conclude this from a single test. For example, some data could also start with 12443 evaluations or with an evaluation which is larger than the max evaluation in another set. I don't see that we have sufficient test coverage to know with reasonable certainty that nothing would break in a variety of cases even in near future.

**Comment by brockho** [comment]
I revised the documentation with a4e0d2f in the `devel-cons` branch and would be fine with closing the issue as this branch is supposed to be merged very soon to `development`, see also #1607.

**Comment by brockho** [comment]
Joint decision of today: can be closed because #1607 will be merged soon.
